File: Juan/P3/Java/Napakalaki/src/NapakalakiGame/TreasureKind.java

package NapakalakiGame;

/**
 *
 * @author juane
 * @version 3.1
 * Enumerado TreasureKind para especificar
 * los tipos de tesoros existentes.
 */
public enum TreasureKind {
     ARMOR, ONEHAND, BOTHHANDS, HELMET, SHOES;
     
     
     @Override
    public String toString() {

        
        if(this.equals(TreasureKind.ARMOR)){
            return "ARMOR";
        }
             
        else if(this.equals(TreasureKind.ONEHAND)){
            return "ONEHAND";
        }
        else if(this.equals(TreasureKind.BOTHHANDS)){
            return "BOTHHANDS";
        }
        else if(this.equals(TreasureKind.HELMET)){
            return "HELMET";
            
        }
        else if(this.equals(TreasureKind.SHOES)){
            return "SHOES";
        }
        
        else
            return "HAY UN ERROR";
        
    }
}

File: Juan/P3/Ruby/Napakalaki/pkg/Napakalaki-0.0.1/lib/TreasureKind.rb
#encoding: utf-8

#Versión 3.0
#Modulo para el tipo enumerado

module NapakalakiGame
module TreasureKind 

    ARMOR = :armor
    ONEHAND = :onehand
    BOTHHANDS = :bothhands
    HELMET = :helmet
    SHOES = :shoes
    
 
  def to_s
      if [TreasureKind.ARMOR]          
          return "ARMOR"
      end
                
          if [TreasureKind.ONEHAND]
             return "ONEHAND"
          end
          
      if [TreasureKind.BOTHHANDS]
          return "BOTHHANDS"
      end
      
      if [TreasureKind.HELMET]
          return "HELMET"
      end
      
      if [TreasureKind.SHOES]
          return "SHOES"
      end
    end
end

end